__init__.py

```diff
@@ -315,13 +298,12 @@ def update_display(self, force=False):
         if self.answering_query:
             return
 
-        self.gui['time_string'] = self.get_display_current_time()
-        self.gui['date_string'] = self.get_display_date()
-        self.gui['ampm_string'] = ''  # TODO
-        self.gui['weekday_string'] = self.get_weekday()
-        self.gui['month_string'] = self.get_month_date()
-
         if self.settings.get("show_time", False):
+            self.gui['time_string'] = self.get_display_current_time()
+            self.gui['date_string'] = self.get_display_date()
+            self.gui['ampm_string'] = ''  # TODO
+            self.gui['weekday_string'] = self.get_weekday()
+            self.gui['month_string'] = self.get_month_date()
```

Comment:
I tossed GUI handling completely here, as this is a MK1 only thing. (repeating event)

I don't know why this had to be forcefully mixed with an adapt intent with the result to add flags like `force` or `show_time` (which btw never resets to `False` when activated.)

Reply:
it's also used if datetime is set as homescreen

Reply:
Not perfectly clear what this means, but i don't seen any connection to homescreen in `update_display`.
Besides, i'm running the refactored skill without it.

And the remaining connections are cut with removing the skill api

Reply:
I must have hallucinated... or we removed it meanwhile and i didnt notice

I could swear this skill provided a clock homescreen, and thats where that code for self.gui came from

Reply:
this https://github.com/MycroftAI/skill-date-time/blob/21.02/__init__.py#LL90C23-L90C23

not sure when it was removed, but i'd like it added back...

Reply:
Ah, looking at the mycroft skill, i think i know where this is coming from

```
    @resting_screen_handler('Time and Date')
    def handle_idle(self, message):
        self.gui.clear()
        self.log.debug('Activating Time/Date resting page')
        self.gui['time_string'] = self.get_display_current_time()
        self.gui['ampm_string'] = ''
        ...
```

Reply:
> not sure when it was removed, but i'd like it added back...

Why is that? I'm not quite familiar with GUI, but shouldn't this kill widgets. or are those overlays?

Reply:
homescreen is configurable, any skill can provide an homescreen, the ovos-homescreen is our reference and most complete one for a mk2, but it doesnt make a lot of sense for non-voice assistants

a simple clock is much less heavy and universal to have available as an option, without even requiring a dedicated homescreen skill. in most install this will be the only available homescreen

Reply:
just that i get the picture. if we put it back in we are still able to override with [idle_display_skill](https://github.com/OpenVoiceOS/ovos-config/blob/129ee42ea55c9b95483aa5f972f3ab435bc1aea4/ovos_config/mycroft.conf#L432), right?

Reply:
yes, the config fields you linked defines the active homescreen, it can also be set via ovos-shell in the dropdown menu, users decide what homescreen they want, and any skill may register itself as a homescreen provider